### PR TITLE
[TAR-995] ADDED a stage for executing License Scans

### PR DESCRIPTION
**Pull request by zelahi**
Signed-off-by: Zuhayr Elahi <elahi.zuhayr@gmail.com>

We are in the process of integrating our license scans to occur pre-commit.  In this PR, we are adding a Jenkinsfile and a Makefile which includes `fossa` commands.  This is to execute a license scan on every commit to check for license violations

The job was created here: 
https://ci.docker.com/teams-dsg/blue/organizations/jenkins/fossa-compose/detail/fossa-compose/2/pipeline

The job ran and a report was uploaded here: https://app.fossa.io/projects/custom+11%2Fgit@github.com:docker%2Fcompose/refs/branch/null/0d061d84d5babeb3403cf4c23fe3c0f79c05002f

Currently it won't run on the same pipeline as the Jenkins that is public facing but we could move it there if the compose team would like us to do so. 

The fossa scans could be flaky due to issues on their end which we are unable to fix, as a result separating them is a nice way for us to handle this as the scans won't block people if they fail

### CODE REVIEW

File: script/Jenkinsfile.fossa [hunk 1]
```diff
@@ -0,0 +1,30 @@
+hubCred = [
+    $class: 'UsernamePasswordMultiBinding',
```
Comment (ndeloof) [inline script/Jenkinsfile.fossa]: $class is legacy syntax to be avoided. 
Also group credential binding within a single `withCredentials` definition

prefer: 
```
withCredentials([
    usernamePassword(credentialsId: 'orcaeng-hub.docker.com', usernameVariable: 'REGISTRY_USERNAME', passwordVariable: 'REGISTRY_PASSWORD'),
    string(credentialsId: 'fossa-api-key', variable: 'FOSSA_API_KEY')
]) { ... }

File: script/fossa.mk [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Variables for Fossa
+BUILD_ANALYZER?=dockereng/fossa-analyzer
+FOSSA_OPTS?=--option all-tags:true --option allow-unresolved:true
+
+fossa-analyze:
+	docker run --rm -e FOSSA_API_KEY=$(FOSSA_API_KEY) \
+		-e GO111MODULE=on \
```
Comment (thaJeztah) [inline script/fossa.mk]: This repository doesn't use Golang, so we probably don't need this line

File: script/fossa.mk [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Variables for Fossa
+BUILD_ANALYZER?=dockereng/fossa-analyzer
+FOSSA_OPTS?=--option all-tags:true --option allow-unresolved:true
```
Comment (thaJeztah) [inline script/fossa.mk]: Does fossa need a "strategy" set for python? https://github.com/fossas/fossa-cli/blob/master/docs/integrations/python.md#python

File: Jenkinsfile [hunk 4]
```diff
@@ -91,5 +91,4 @@ baseImages.each { baseImage ->
     }
   }
 }
-
```
Comment (thaJeztah) [inline Jenkinsfile]: nit; unrelated change now
